### PR TITLE
[TASK] Adjust autoloading documentation for TYPO3 8.7

### PR DESCRIPTION
**Pull request by alexanderschnitzler**
Same content that has been added for 7.6 already.

### CODE REVIEW

**Comment by alexanderschnitzler** [comment]
Do not merge before https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/pull/149 is final and merged. I'll update this PR again.

File: Documentation/ApiOverview/Autoloading/Index.rst [hunk 1]
```diff
@@ -13,104 +14,81 @@ and :ref:`XCLASS <xclasses>` handling.
...
-with the class name as key and the file location as value. No other code is allowed in this file.
+:ref:`ObjectManager <t3cmsapi:TYPO3\\CMS\\Extbase\\Object\\ObjectManager>`
+(which internally uses :php:`makeInstance()` again).
 
+.. _autoloading_since_typo3_7:
+Autoloading classes since TYPO3 7.x
+===================================
 
-.. _autoload-examples:
+TYPO3 6.2 was still delivered with a couple of different autoloaders, that all had different approaches and rules to find a class. This led to the naming conventions in and outside extbase and the optional `ext_autoload.php` file to load classes that didn't follow the conventions. Since TYPO3 7.0 all this is gone and there is only a single autoloader left, the one of composer. No matter if you run TYPO3 in composer mode or not, TYPO3 uses the composer autoloader to resolve all class file locations. However, the autoloader is little bit more sophisticated in composer mode as it then supports `PSR-4` autoloading.
```
Comment (liayn) [inline Documentation/ApiOverview/Autoloading/Index.rst]: > However, the autoloader is little bit more sophisticated in composer mode as it then supports `PSR-4` autoloading.
That's not true. It always supports PSR-4, independent of composer mode

File: Documentation/ApiOverview/Autoloading/Index.rst [hunk 2]
```diff
@@ -13,104 +14,81 @@ and :ref:`XCLASS <xclasses>` handling.
...
 
-Examples for non-namespaced classes
-===================================
+.. _autoloading_without_composer_mode:
+Loading classes without composer mode
+=====================================
 
-The examples below are related to non-namespaced classes. When using :ref:`namespaces <namespaces>`,
-autoloading will happen without any extra effort on your part.
+This means, you did not install TYPO3 via a require-statement inside your composer.json. It's a regular old-school install where the TYPO3 source and the symlinks (`typo3/index.php`) are setup manually. In this case, every time you install an extension, the autoloader scans the whole extension directory for classes. No matter if they follow any convention at all. There is just one rule. Put each class into its own file. The generated classmap is a huge array with a mapping of classnames to their location on the disk.
```
Comment (liayn) [inline Documentation/ApiOverview/Autoloading/Index.rst]: >  No matter if they follow any convention at all. 
This is wrong too.

File: Documentation/ApiOverview/Autoloading/Index.rst (hunk 1 above)
Comment (alexanderschnitzler) [inline Documentation/ApiOverview/Autoloading/Index.rst]: That's why I wrote, that #149 should be merged before. Let's focus on that PR and simply use the same version for latest then.

**Comment by lolli42** [comment]
Will merge now since i already merged the 7.6 version like that. if that needs further improvements Markus, please go ahead.

**Pull request merged**